homeassistant/components/application_credentials/manifest.json

```diff
@@ -0,0 +1,9 @@
+{
+  "domain": "application_credentials",
+  "name": "Application Credentials",
+  "config_flow": false,
+  "documentation": "https://www.home-assistant.io/integrations/application_credentials",
+  "dependencies": ["http"],
```

Comment:
Why does it depend on `http` ?

Reply:
It needs the OAuth HTTP callback APIs. My impression was that integrations should be responsible for the HTTP APIs provided the OAuth helpers since OAuth config flow is not a component, rather than hoping for the best.

Is it a better practice to assume that some other component will register this and just make it a test only dependency?

```
ERROR    homeassistant.setup:setup.py:245 Error during setup of component application_credentials
Traceback (most recent call last):
  File "/home/allen/home-assistant-core/homeassistant/setup.py", line 235, in _async_setup_component
    result = await task
  File "/home/allen/home-assistant-core/homeassistant/components/application_credentials/__init__.py", line 127, in async_setup
    config_entry_oauth2_flow.async_register_local_apis(hass)
  File "/home/allen/home-assistant-core/homeassistant/helpers/config_entry_oauth2_flow.py", line 349, in async_register_local_apis
    hass.http.register_view(OAuth2AuthorizeCallbackView())
AttributeError: 'NoneType' object has no attribute 'register_view'
```

Reply:
Good point. Yeah it should depend on it.

Or actually, we should register `OAuth2AuthorizeCallbackView` maybe in the `auth` component? It's weird that it's dynamically loaded.

Reply:
I'll "pre-factor" this into a separate PR right now.

Reply:
Sent https://github.com/home-assistant/core/pull/69507

Reply:
I rebased and removed the changes around http registration, and updated this to depend on auth. All the complexity around registration is gone now.